app/main.py

```diff
@@ -2,0 +1,42 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self._hidden: bool = False
+        Animal.alive.append(self)
+
+    @property
+    def hidden(self) -> bool:
+        return self._hidden
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+
+class Herbivore(Animal):
+    def __sub__(self, damage: int) -> None:
+        self.health = max(0, self.health - damage)
+        if self.health == 0:
+            self.die()
+
+    def hide(self) -> None:
+        self._hidden = not self._hidden
+
+
+class Carnivore(Animal):
```

Comment:
Code Efficiency: You should use `Carnivore` instead of `self` in the `bite` method. Also, the `bite` method should not be a static method because it is an action performed by an instance of a Carnivore.